### DISCUSSION TITLE
Different time zones (wait module, 'relative to')

### DISCUSSION THREAD

**Discussion by alpkaanaksu** [rubenarslan/formr.org#554] [Q&A]
We will use formr for an international study, we expect to have a lot of people from the US and Germany.

We ask participants to pick a time to receive daily evening surveys, and use the chosen time in the wait modules.

Would this work with different time zones?

**Comment by rubenarslan** [comment]
Yes, it's possible. Unfortunately, you'll have to do the timezone calculations yourself, formr schedules everything in German time. But many have done this. You have to collect time zone information. Then, I'd suggest using lubridate's time zone functions for scheduling, so you don't accidentally mess up stuff like daylight savings time. https://lubridate.tidyverse.org/reference/with_tz.html

**Reply by alpkaanaksu** [reply]
> You have to collect time zone information.

Should we ask the participants? Or is there a way to get it from the browser by injecting JavaScript. Can we change the results by adding custom JavaScript?

It could work, if we can save the value of the following:

```js
Intl.DateTimeFormat().resolvedOptions().timeZone 
```

**Reply by rubenarslan** [reply]
Hi Alp. I think combined is best. Ask, but make it default to the browser setting (which may or may not be correct). See attached JSON for a minimal implementation. Maybe we should make this behaviour the default in formr.
[timezone.json](https://github.com/user-attachments/files/15887438/timezone.json)
